activerecord/test/cases/associations/has_many_associations_test.rb

```diff
@@ -478,6 +478,14 @@ def test_taking_not_found
     assert_raise(ActiveRecord::RecordNotFound) { authors(:bob).posts.take! }
   end
 
+  def test_taking_with_a_number
+    assert_equal [posts(:misc_by_bob)], authors(:bob).posts.take(1)
+    assert_equal [posts(:misc_by_bob), posts(:other_by_bob)], authors(:bob).posts.take(2)
+    authors(:bob).posts.to_a
```

Comment:
This doesn't address my comment. If you revert this line https://github.com/rails/rails/pull/19105/files#diff-20f545c453ee24942b6f7ae565e9e369R136 your test will still pass.